lib/passport-azure-ad/metadata.js

```diff
@@ -102,41 +121,128 @@ Metadata.prototype.updateWsfedMetadata = function(doc, next) {
   }
 };
 
+Metadata.prototype.updateOidcMetadata = function(doc, next) {
+  log.info('Request to update the Open ID Connect Metadata');
+  try {
+    this.oidc = {};
+    this.oidc.issuer = doc['issuer'];
+    this.oidc.algorithms = doc['id_token_signing_alg_values_supported'];
+    var jwksUri = doc.jwks_uri;
+
+    log.info('Algorithm retreived was: ', this.oidc.algorithms);
+    log.info('Issuer we are using is: ', this.oidc.issuer);
+    log.info('Key Endpoint will we use is: ', jwksUri);
+
+    var self = this;
+    var callback = next;
+
+    async.waterfall([
+      // fetch the signing keys
+      function(next) {
+        request(jwksUri, function(err, response, body) {
+          if (err) {
+            next(err);
+          } else if (response.statusCode !== 200) {
+            next(new Error("Error:" + response.statusCode + " Cannot get AAD Signing Keys"));
+          } else {
+            next(null, body);
+          }
+        });
+      },
+
+      function(body, next) {
+        // parse the AAD Federation metadata xml
+        log.info("Parsing JSON retreived from the signing keys endpoint.");
+        try {
+          self.oidc.keys = JSON.parse(body).keys;
+          log.info("***** KEYS  ******");
+          log.info("");
+          log.info(self.oidc.keys);
+          log.info("");
+          log.info("***********");
+          next(null);
+        } catch (e) {
+          next(new Error(e));
+        }
+      },
+
+    ], function(err) {
+      callback(err);
+    });
+
+  } catch (e) {
+    next(new Error('Invalid Open ID Connect Federation Metadata ' + e.message));
+  }
+};
+
+Metadata.prototype.generateOidcPEM = function(kid) {
+  if (!this.oidc.keys) {
+    return null;
+  }
+  for (var i = 0; i < this.oidc.keys.length; i++) {
+    if (this.oidc.keys[i].kid === kid) {
+      log.info('Working on key: ', this.oidc.keys[i]);
```

Comment:
this always assumes e, n. what happens with ONLY x5c.  Should we error and log in that case?